sled-agent/src/sled_agent.rs

```diff
@@ -1121,6 +1124,31 @@ impl SledAgent {
         &self.inner.boot_disk_os_writer
     }
 
+    /// Return identifiers for this sled.
+    ///
+    /// This is mostly used to identify timeseries data with the originating
+    /// sled.
+    ///
+    /// NOTE: This only returns the identifiers for the _sled_ itself. If you're
+    /// interested in the switch identifiers, MGS is the current best way to do
+    /// that, by asking for the local switch's slot, and then that switch's SP
+    /// state.
+    pub(crate) async fn sled_identifiers(
+        &self,
+    ) -> Result<SledIdentifiers, Error> {
+        let baseboard = self.inner.hardware.baseboard();
+        Ok(SledIdentifiers {
+            rack_id: self.inner.start_request.body.rack_id,
+            sled_id: self.inner.id,
+            model: baseboard.model().to_string(),
+            revision: baseboard
+                .revision()
+                .try_into()
+                .map_err(|_| Error::UnexpectedRevision(baseboard.revision()))?,
```

Comment:
The revision is an `i64` -- this should always be translatable to a string, right? Can't we just use `to_string()`? or is there a situation where this error could actually happen?

Reply:
oh, or is this because we're going from i64 -> u32? It's a little surprising these don't align -- what type do we want our revision to be?

Reply:
It's the latter, converting `i64` to `u32`. The actual devinfo property we read this from is a `int64_t`, but I was under the impression the values are actually `u32`s. I'm still looking for the software that actually populates the revision though. I'll report back.

Reply:
Here it is. Below is the definition of the identifying data, fetched from the SP via IPCC and filled in during early boot.

https://github.com/oxidecomputer/illumos-gate/blob/e517f5ad5a166aedb4450aee69f8b8e80cd5a8c1/usr/src/uts/oxide/sys/ipcc.h#L68-L72

That revision is a `uint32_t`, so we can't really hit this error condition until / unless we change that interface.

It might make sense to change the revision in `Baseboard::Gimlet` to a `u32`, I'm not entirely sure. That would probably be more disruptive, since much of the code is currently infallible and this would introduce a fallible path in a few places. Thoughts?

Reply:
It's also a `u32` from the SP's side:

https://github.com/oxidecomputer/hubris/blob/fca59bdbd6e83d2fc31c7f1a14b28de463e2095b/lib/oxide-barcode/src/lib.rs#L40-L43

I filed https://github.com/oxidecomputer/omicron/issues/6095 to track changing the revision to a `u32`, which would remove this error path.